packages/core/admin/admin/src/components/AuthenticatedApp/index.js

```diff
@@ -0,0 +1,70 @@
+import React, { useEffect, useState } from 'react';
+import { LoadingIndicatorPage, AppInfosContext } from '@strapi/helper-plugin';
+import { useQueries } from 'react-query';
+import packageJSON from '../../../../package.json';
+import PluginsInitializer from '../PluginsInitializer';
+import RBACProvider from '../RBACProvider';
+import { fetchAppInfo, fetchCurrentUserPermissions, fetchStrapiLatestRelease } from './utils/api';
+import checkLatestStrapiVersion from './utils/checkLatestStrapiVersion';
+
+const { STRAPI_ADMIN_UPDATE_NOTIFICATION } = process.env;
+const canFetchRelease = STRAPI_ADMIN_UPDATE_NOTIFICATION === 'true';
+const strapiVersion = packageJSON.version;
+
+const AuthenticatedApp = () => {
+  const [state, setState] = useState({
+    shouldUpdateStrapi: false,
+    latestStrapiReleaseTag: strapiVersion,
+  });
+
+  const [
+    { data: appInfos, status },
+    { data: tag_name, status: releaseStatus, isLoading },
+    { data: permissions, status: fetchPermissionsStatus, refetch, isFetched, isFetching },
+  ] = useQueries([
+    { queryKey: 'app-infos', queryFn: fetchAppInfo },
+    {
+      queryKey: 'strapi-release',
+      queryFn: fetchStrapiLatestRelease,
+      enabled: canFetchRelease,
+      initialData: strapiVersion,
+    },
+    {
+      queryKey: 'admin-users-permission',
+      queryFn: fetchCurrentUserPermissions,
+      initialData: [],
+    },
+  ]);
+
+  useEffect(() => {
+    if (releaseStatus === 'success') {
+      const shouldUpdateStrapi = checkLatestStrapiVersion(strapiVersion, tag_name);
+
+      setState({ shouldUpdateStrapi, latestStrapiReleaseTag: tag_name });
+    }
+  }, [releaseStatus, tag_name]);
```

Comment:
Not sure this is necessary. I think we might just derivate the state during the render:

```javascript
const shouldUpdateStrapi = releaseStatus === 'success' ? checkLatestStrapiVersion(strapiVersion, tag_name) : undefined;

const latestStrapiReleaseTag = releaseStatus === 'success' ? tag_name : undefined;
```

outside the useEffect no?

Reply:
YEs but if we do it outside the effect we will compute on every render, I think it is not necessary.

Reply:
Does it have a real impact?

Reply:
It renders on every url change so a lot...